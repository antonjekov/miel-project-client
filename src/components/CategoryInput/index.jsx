import React from "react";
import styles from './index.module.css';
import { Form } from 'react-bootstrap';

function CategoryInput({ values, errors, handleChange,categoryOptions }) {

    return (
        <Form.Group controlId="category">
            <Form.Label >Category</Form.Label>
            <Form.Control className={styles['form-control']} as="select" value={values.category} name='category' onChange={handleChange} isInvalid={!!errors.category} isValid={values.category && !errors.category}>
                <option></option>
                {categoryOptions}
            </Form.Control>
            <Form.Control.Feedback type='invalid' >{errors.category}</Form.Control.Feedback>
            <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
        </Form.Group>
    )
}

export default CategoryInput