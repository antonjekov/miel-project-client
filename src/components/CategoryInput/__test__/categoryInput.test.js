import React from 'react'
import renderer from "react-test-renderer"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import CategoryInput from '..'

describe('CategoryInput', ()=>{

    afterEach(cleanup)
    it('matches snapshot without error', () => {
        const values = {category: 'honey'}
        const errors = {}
        const tree = renderer.create( <CategoryInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
    it('matches snapshot with error', () => {
        const values = {category: 'honey'}
        const errors = {category: 'Some error'}
        const tree = renderer.create( <CategoryInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })

})
