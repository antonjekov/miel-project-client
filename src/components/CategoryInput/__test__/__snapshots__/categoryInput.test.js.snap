// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryInput matches snapshot with error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="category"
  >
    Category
  </label>
  <select
    className="form-control form-control is-invalid"
    id="category"
    name="category"
    onChange={[Function]}
    value="honey"
  >
    <option />
  </select>
  <div
    className="invalid-feedback"
  >
    Some error
  </div>
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;

exports[`CategoryInput matches snapshot without error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
    htmlFor="category"
  >
    Category
  </label>
  <select
    className="form-control form-control is-valid"
    id="category"
    name="category"
    onChange={[Function]}
    value="honey"
  >
    <option />
  </select>
  <div
    className="invalid-feedback"
  />
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;
